src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Carnivore) {
+      return;
+    }
+
+    if (!animal.hidden) {
+      animal.health -= 50;
+    }
```

Comment:
let's move this into one block 
```suggestion
    if (animal instanceof Herbivore && !animal.hidden) {
      animal.health -= 50;
    }
```

Reply:
Thanks, it looks better